distributed/protocol/serialize.py

```diff
@@ -783,7 +785,14 @@ def _deserialize_memoryview(header, frames):
         out = ensure_memoryview(frames[0])
     else:
         out = memoryview(b"".join(frames))
-    out = out.cast(header["format"], header["shape"])
+
+    # handle empty `memoryview`s
+    if out:
+        out = out.cast(header["format"], header["shape"])
+    else:
+        out = out.cast(header["format"])
+        assert out.shape == header["shape"]
```

Comment:
Is this assertion necessary ?  I'm thinking about what happens if a user gets this ?  Is the intent to inform the user something in serializing/deserializing has gone awry ?

Reply:
Yeah the `.cast` effectively does this check for us. However it is not handled in the `else` case so added an `assert`

If this assertion raises, it means something has gone horribly wrong. For example we expected a non-empty `memoryview`, but got an empty one. IOW data was lost in communication somewhere.

Reply:
Based off offline discussion, it sounds like this is ok. Though happy to discuss further if needed